tools/ci/testlist/sim-01.dat

```diff
@@ -6,16 +6,19 @@
 -Darwin,sim:cxxtest
 -Darwin,sim:libcxxtest
 
-# Do not build Linux configs
--Darwin,sim:linuxi2c
--Darwin,sim:linuxspi
+# clang doesn't -fsanitize=kernel-address
+-Darwin,sim:kasan
```

Comment:
Is this really true? Or this lack of support is just that it has not been verified to work with NuttX?
Clang seems to support kernel address sanitizers for a long time.

Reply:
At least, clang on our macOS CI can't recognize -fsanitize=kernel-address. Here is the link before I add kasan to the black list(https://github.com/apache/incubator-nuttx/runs/4055120804?check_suite_focus=true):
```
clangclang: error: error: clangunsupported option '-fsanitize=kernel-address' for target 'x86_64-apple-darwin19.6.0': : error: unsupported option '-fsanitize=kernel-address' for target 'x86_64-apple-darwin19.6.0'

unsupported option '-fsanitize=kernel-address' for target 'x86_64-apple-darwin19.6.0'
```

Reply:
It seems that the `LLVM/clang` that ships with macOS somehow does not support sanitizers: https://stackoverflow.com/questions/65112164/how-to-resolve-clang-error-unsupported-option-fsanitize-leak-on-apple-dev

Let's address it later. Probably we would need to install the host toolchain (either clang or GCC) from Homwbrew to make it work properly.